File: LibraryManagement.Application/DTOs/Book/BookResponseDto.cs
// namespace LibraryManagement.API.DTOs.Book;
// public class BookResponseDto
// {
//     public Guid Id { get; set; }
//     public string Title { get; set; } = string.Empty;
//     public string? Isbn { get; set; }
//     public int? PublishedYear { get; set; }

//     public string? CoverImageUrl { get; set; }

//     public string? SubCategoryName { get; set; }

//     public List<string> Authors { get; set; } = new();
//     public List<string> Categories { get; set; } = new();

//     public int TotalCopies { get; set; }
// }



using LibraryManagement.API.DTOs.Author;
using LibraryManagement.API.DTOs.Copy;

namespace LibraryManagement.API.DTOs.Book;

public class BookResponseDto
{
    public Guid Id { get; set; }
    public string Title { get; set; } = string.Empty;

    public string? Isbn { get; set; }
    public int? PublishedYear { get; set; }

    public string? CoverImageUrl { get; set; }
    public string? SubCategoryName { get; set; }
    public List<AuthorResponseDto>? Authors { get; set; }
    public List<string> Categories { get; set; } = new();

    public string? Language { get; set; }
    public string? Summary { get; set; }


    public List<CopyResponseDto> Copies { get; set; } = new();

    public int TotalCopies { get; set; }

    public int AvailableCopies { get; set; }
}

File: Frontend/library-app/src/app/core/models/book-response.dto.ts
import { AuthorResponseDto } from "./author-response.dto";
import { CopyDto } from "./copy.dto";
import { SubCategoryResponseDto } from "./subcategory-response.dto";



export interface BookResponseDto {
  SubCategoryResponseDto: any;
  id: string;
  title: string;

  isbn?: string;
  publishedYear?: number;
  coverImageUrl?: string;

  subCategory: SubCategoryResponseDto;

  authors: AuthorResponseDto[]; 
  categories: string[];

  language?: string;
  summary?: string;

  copies: CopyDto[];

  totalCopies: number;

  availableCopies?: number;

  status:boolean;
}